warehouse/query-core/src/test/java/datawave/query/tables/RangeStreamScannerTest.java

```diff
@@ -0,0 +1,374 @@
+     */
+    @Test
+    public void testExceedMaxMedianDocumentsPerShard() throws Exception {
+        
+        // Components that define the query: "FOO == 'boo'"
+        String fieldName = "FOO";
+        String fieldValue = "boo";
+        ASTEQNode eqNode = (ASTEQNode) JexlNodeFactory.buildEQNode(fieldName, fieldValue);
+        
+        // Construct a ScannerStream from RangeStreamScanner, iterator, entry parser.
+        RangeStreamScanner rangeStreamScanner = buildRangeStreamScanner(fieldName, fieldValue);
+        EntryParser entryParser = new EntryParser(eqNode, fieldName, fieldValue, config.getIndexedFields());
+        Iterator<Tuple2<String,IndexInfo>> iterator = Iterators.transform(rangeStreamScanner, entryParser);
+        ScannerStream scannerStream = ScannerStream.initialized(iterator, eqNode);
+        
+        // Assert the iterator correctly iterates over the iterables without irritating the unit test.
+        assertTrue(scannerStream.hasNext());
+        int shardCount = 0;
+        int documentCount = 0;
+        while (scannerStream.hasNext()) {
+            Tuple2<String,IndexInfo> entry = scannerStream.next();
+            assertTrue(entry.first().startsWith("20190319_"));
+            assertEquals(15, entry.second().count());
+            shardCount++;
+            documentCount += entry.second().count();
+        }
+        assertEquals(8, shardCount);
+        assertEquals(120, documentCount);
+        assertFalse(scannerStream.hasNext());
+    }
+    
+    /**
+     * FOO == 'boohoo' hits day 20190319 with 15 shards, each shard has 25 document ids.
+     */
+    @Test
+    public void testExceedShardsPerDayThresholdAndDocumentsPerShardThreshold() throws Exception {
+        
+        // Components that define the query: "FOO == 'boohoo'"
+        String fieldName = "FOO";
+        String fieldValue = "boohoo";
+        ASTEQNode eqNode = (ASTEQNode) JexlNodeFactory.buildEQNode(fieldName, fieldValue);
+        
+        // Construct a ScannerStream from RangeStreamScanner, iterator, entry parser.
+        RangeStreamScanner rangeStreamScanner = buildRangeStreamScanner(fieldName, fieldValue);
+        EntryParser entryParser = new EntryParser(eqNode, fieldName, fieldValue, config.getIndexedFields());
+        Iterator<Tuple2<String,IndexInfo>> iterator = Iterators.transform(rangeStreamScanner, entryParser);
+        ScannerStream scannerStream = ScannerStream.initialized(iterator, eqNode);
+        
+        // Assert the iterator correctly iterates over the iterables without irritating the unit test.
+        assertTrue(scannerStream.hasNext());
+        int shardCount = 0;
+        int documentCount = 0;
+        while (scannerStream.hasNext()) {
+            Tuple2<String,IndexInfo> entry = scannerStream.next();
+            assertTrue("Expected shard to start with '20190323' but was: " + entry.first(), entry.first().startsWith("20190323"));
```

Comment:
Do we want to add this verbiage for this check in each of the tests?  It might be useful.

Reply:
I'll add this to the other tests so it's consistent